src/colmap/geometry/pose.cc

```diff
@@ -153,6 +154,27 @@ Rigid3d InterpolateCameraPoses(const Rigid3d& cam_from_world1,
                  cam_from_world1.translation + translation12 * t);
 }
 
+Eigen::Quaterniond QuaternionFromAngleAxis(const Eigen::Vector3d& omega) {
+  double half_angle = 0.5 * omega.norm();
+  Eigen::Vector3d axis = omega.normalized();
+  Eigen::Quaterniond q(Eigen::AngleAxisd(half_angle, axis));
+  return q;
+}
+
+Eigen::Matrix3d RightJacobianFromAngleAxis(const Eigen::Vector3d& omega) {
+  Eigen::Matrix3d skew_omega = CrossProductMatrix(omega);
+  double theta = omega.norm();
+  Eigen::Matrix3d J;
+  if (theta < 1e-6)
+    J = Eigen::Matrix3d::Identity() - 0.5 * skew_omega;
+  else {
+    Eigen::Matrix3d M = skew_omega / theta;
+    J = Eigen::Matrix3d::Identity() - ((1.0 - std::cos(theta)) / theta) * M +
+        (1.0 - std::sin(theta) / theta) * M * M;
+  }
+  return J;
```

Comment:
```suggestion
  const double theta = omega.norm();
  if (theta < 1e-6) {
    return Eigen::Matrix3d::Identity() - 0.5 * skew_omega;
  } else {
    const Eigen::Matrix3d M = skew_omega / theta;
    return Eigen::Matrix3d::Identity() - (((1.0 - std::cos(theta)) / theta) +
        (1.0 - std::sin(theta) / theta) * M) * M;
  }
```

Reply:
Thanks. (1.0 - std::cos(theta)) / theta) is a double so we need to multiple an identity matrix here. This is resolved.